src/main/java/io/confluent/connect/elasticsearch/ElasticsearchClient.java

```diff
@@ -658,6 +658,12 @@ private void removeFromInFlightRequests(long executionId) {
    */
   private synchronized void reportBadRecord(BulkItemResponse response,
                                             long executionId) {
+
+    // RCCA-7507 : Don't push to DLQ if we receive Internal version conflict on data streams
+    if (config.isDataStream()) {
```

Comment:
Should this condition be - if it is datastream AND encountered version conflict error.